### PR TITLE
Remove haystack from code base

### PR DESCRIPTION
**Pull request by stsewd**
Ok, this is a big remove (I'm more worried more about the apiv1 and some templates).

This closes #4006

RTD doesn't use haystack on production anymore. Also, I try some endpoints

- https://readthedocs.org/api/v1/file/search/?format=json&q=virtualenvwrapper
- https://readthedocs.org/api/v1/project/search/?q=rtd&format=json

Both give a 500 error code.

The only part that haystack was really being used was on some templates to use the `highlight` https://django-haystack.readthedocs.io/en/master/templatetags.html#highlight tag, so, I'm not sure if it's good to remove this from here since it gives a nice visual hint for the user, maybe we can search another lib that gives the same functionality or maybe this isn't so relevant and we are ok removing it.

Note: I wasn't able to full tests this on the search page with results because I don't have elasticsearch, I will try to setup elasticsearch to test this later.

### CODE REVIEW

File: readthedocs/settings/base.py [hunk 1]
```diff
@@ -84,9 +84,6 @@ def INSTALLED_APPS(self):  # noqa
             'annoying',
             'django_extensions',
             'messages_extends',
-
-            # daniellindsleyrocksdahouse
```
Comment (stsewd) [inline readthedocs/settings/base.py]: Weird comment here :grin:

File: readthedocs/templates/search/base_search.html [hunk 2]
```diff
@@ -15,7 +14,7 @@
           <a href="{% if result.absolute_url %}{{ result.absolute_url }}{% else %}{{ result.object.get_absolute_url }}{% endif %}?highlight={{ query }}">{{ result.project }} - {{ result.title }}</a>
         </p>
         <p>
-          {% highlight result.text with query %}
+          {% result.text %}
```
Comment (stsewd) [inline readthedocs/templates/search/base_search.html]: We are losing some visual hints here

File: readthedocs/templates/search/faceted_results.html [hunk 3]
```diff
@@ -55,7 +55,7 @@ <h5>{% trans "Filter by Project" %}</h5>
                 <a href="{% if result.absolute_url %}{{ result.absolute_url }}{% else %}{{ result.object.get_absolute_url }}{% endif %}?highlight={{ query }}">{{ result.project }} {% if result.version %} ({{ result.version }}) {% endif %} - {{ result.title }}</a>
               </p>
               <p>
-                {% highlight result.text with query %}
+                {% result.text %}
```
Comment (stsewd) [inline readthedocs/templates/search/faceted_results.html]: ...and here

File: requirements/pip.txt [hunk 4]
```diff
@@ -23,8 +23,6 @@ readthedocs-build<2.1
 # django-tastypie 0.13.x and 0.14.0 are not compatible with our code
 django-tastypie==0.13.0
 
-django-haystack==2.8.1
-celery-haystack==0.10
```
Comment (stsewd) [inline requirements/pip.txt]: I'm not sure where the celery-haystack was used, I assume it was some dependency from django-haystack

**Comment by stsewd** [comment]
Travis is all green, that's a good sign :green_apple:

**Comment by ericholscher** [comment]
ES is not used for all highlighting of search results. The entire set of search templates can also be deleted, as our search results use `elastic_search.html` now: https://github.com/rtfd/readthedocs.org/blob/master/readthedocs/search/views.py#L94

**Comment by stsewd** [comment]
I deleted the templates that were used previously, I checked the search pages and those work correctly and also I checked some endpoints and those are good too.

**Comment by ericholscher** [comment]
Looks good. I'll give this a spin locally as well, but I don't forsee any issues since haystack hasn't been used in _years_.

**Comment by ericholscher** [comment]
Looks good. Wee!

**Pull request merged**